osf/metrics/reporters/institution_summary_monthly.py

```diff
@@ -0,0 +1,93 @@
+            published_preprint_count=self.get_published_preprints(institution).count(),
+            storage_byte_count=self.get_storage_size(node_queryset, institution),
+            public_file_count=self.get_files(node_queryset, institution, is_public=True).count(),
+            monthly_logged_in_user_count=self.get_monthly_logged_in_user_count(institution, yearmonth),
+            monthly_active_user_count=self.get_monthly_active_user_count(institution, yearmonth),
+        )
+
+    def _get_count(self, node_queryset, node_type, is_public):
+        return node_queryset.filter(type=node_type, is_public=is_public, root_id=F('pk')).count()
+
+    def get_published_preprints(self, institution):
+        return Preprint.objects.can_view().filter(
+            affiliated_institutions=institution
+        ).exclude(spam_status=SpamStatus.SPAM)
+
+    def get_files(self, node_queryset, institution, is_public=None):
+        public_kwargs = {}
+        if is_public:
+            public_kwargs = {'is_public': is_public}
+
+        target_node_q = Q(
+            target_object_id__in=node_queryset.filter(**public_kwargs).values('pk'),
+            target_content_type=ContentType.objects.get_for_model(AbstractNode),
+        )
+        target_preprint_q = Q(
+            target_object_id__in=self.get_published_preprints(institution).values('pk'),
+            target_content_type=ContentType.objects.get_for_model(Preprint),
+        )
+        return OsfStorageFile.objects.filter(
+            deleted__isnull=True, purged__isnull=True
+        ).filter(target_node_q | target_preprint_q)
+
+    def get_storage_size(self, node_queryset, institution):
+        files = self.get_files(node_queryset, institution)
+        return FileVersion.objects.filter(
+            size__gt=0,
+            purged__isnull=True,
+            basefilenode__in=files
+        ).aggregate(storage_bytes=Sum('size', default=0))['storage_bytes']
+
+    def get_month_start_end(self, yearmonth):
+        # Get the first day of the month
+        start_date = datetime(yearmonth.year, yearmonth.month, 1)
+        # Calculate the first day of the next month
+        if yearmonth.month == 12:
+            end_date = datetime(yearmonth.year + 1, 1, 1)
+        else:
+            end_date = datetime(yearmonth.year, yearmonth.month + 1, 1)
+        return start_date, end_date
+
+    def get_monthly_logged_in_user_count(self, institution, yearmonth):
+        start_date, end_date = self.get_month_start_end(yearmonth)
+        return institution.get_institution_users().filter(
+            date_last_login__gte=start_date,
+            date_last_login__lt=end_date
+        ).count()
+
+    def get_monthly_active_user_count(self, institution, yearmonth):
+        start_date, end_date = self.get_month_start_end(yearmonth)
+        return institution.get_institution_users().filter(
+            date_disabled__isnull=True,
```

Comment:
is "active" supposed to mean only that the account has not been deactivated? not seeing much point to this measure, except to indirectly show how many of this institution's users have been banned this month (by subtracting from the "logged-in" count)

Reply:
Yeah I see what you mean, but I think it should just be a running total of non-deleted users, that makes more sense then user logins, together the numbers give a lot of context.